frame/support/src/storage/bounded_vec.rs

```diff
@@ -39,60 +44,60 @@ use sp_std::{marker::PhantomData, prelude::*};
+pub struct BoundedVec<T, B, S: Get<B>>(Vec<T>, PhantomData<B>, PhantomData<S>);
 
 /// A bounded slice.
 ///
 /// Similar to a `BoundedVec`, but not owned and cannot be decoded.
 #[derive(Encode, scale_info::TypeInfo)]
 #[scale_info(skip_type_params(S))]
-pub struct BoundedSlice<'a, T, S>(&'a [T], PhantomData<S>);
+pub struct BoundedSlice<'a, T, B, S: Get<B>>(&'a [T], PhantomData<B>, PhantomData<S>);
 
 // `BoundedSlice`s encode to something which will always decode into a `BoundedVec`,
 // `WeakBoundedVec`, or a `Vec`.
-impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<BoundedVec<T, S>> for BoundedSlice<'a, T, S> {}
-impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<WeakBoundedVec<T, S>>
-	for BoundedSlice<'a, T, S>
+impl<'a, T: Encode + Decode, B: AtMost32BitUnsigned, S: Get<B>> EncodeLike<BoundedVec<T, B, S>> for BoundedSlice<'a, T, B, S> {}
+impl<'a, T: Encode + Decode, B: AtMost32BitUnsigned, S: Get<B>> EncodeLike<WeakBoundedVec<T, S>>
+	for BoundedSlice<'a, T, B, S>
 {
 }
-impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<Vec<T>> for BoundedSlice<'a, T, S> {}
+impl<'a, T: Encode + Decode, B: AtMost32BitUnsigned, S: Get<B>> EncodeLike<Vec<T>> for BoundedSlice<'a, T, B, S> {}
 
-impl<T: PartialOrd, Bound: Get<u32>> PartialOrd for BoundedVec<T, Bound> {
+impl<T: PartialOrd, B: AtMost32BitUnsigned, Bound: Get<B>> PartialOrd for BoundedVec<T, B, Bound> {
 	fn partial_cmp(&self, other: &Self) -> Option<sp_std::cmp::Ordering> {
 		self.0.partial_cmp(&other.0)
 	}
 }
 
-impl<T: Ord, Bound: Get<u32>> Ord for BoundedVec<T, Bound> {
+impl<T: Ord, B: AtMost32BitUnsigned, Bound: Get<B>> Ord for BoundedVec<T, B, Bound> {
 	fn cmp(&self, other: &Self) -> sp_std::cmp::Ordering {
 		self.0.cmp(&other.0)
 	}
 }
 
-impl<'a, T, S: Get<u32>> TryFrom<&'a [T]> for BoundedSlice<'a, T, S> {
+impl<'a, T, B: AtMost32BitUnsigned, S: Get<B>> TryFrom<&'a [T]> for BoundedSlice<'a, T, B, S> {
 	type Error = ();
 	fn try_from(t: &'a [T]) -> Result<Self, Self::Error> {
-		if t.len() < S::get() as usize {
-			Ok(BoundedSlice(t, PhantomData))
+		if S::get().into().try_into().map_or(false, |v: usize| t.len() < v) {
+			Ok(BoundedSlice(t, PhantomData, PhantomData))
 		} else {
 			Err(())
 		}
 	}
 }
 
-impl<'a, T, S> From<BoundedSlice<'a, T, S>> for &'a [T] {
-	fn from(t: BoundedSlice<'a, T, S>) -> Self {
+impl<'a, T, B, S: Get<B>> From<BoundedSlice<'a, T, B, S>> for &'a [T] {
+	fn from(t: BoundedSlice<'a, T, B, S>) -> Self {
 		t.0
 	}
 }
 
-impl<T: Decode, S: Get<u32>> Decode for BoundedVec<T, S> {
+impl<T: Decode, B: AtMost32BitUnsigned, S: Get<B>> Decode for BoundedVec<T, B, S> {
 	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
 		let inner = Vec::<T>::decode(input)?;
-		if inner.len() > S::get() as usize {
+		if S::get().into().try_into().map_or(false, |v: usize| inner.len() > v) {
 			return Err("BoundedVec exceeds its limit".into())
 		}
-		Ok(Self(inner, PhantomData))
+		Ok(Self(inner, PhantomData, PhantomData))
```

Comment:
```suggestion
		Ok(Self(inner, PhantomData))
```

Reply:
Replacing `Ok(Self(inner, PhantomData, PhantomData))` with `Ok(Self(inner, PhantomData))` gives me a compiler error of

`expected 3 arguments, found 2`,

I think I haven't understood the suggestion. Can we somehow drop the number of type parameters using this approach ?

Reply:
Issue solved